app/helpers/ledger.js

```diff
@@ -0,0 +1,175 @@
+import { wallet } from "wallet-preload-shim";
+import { hexToBytes, strHashToRaw } from "helpers";
+import * as secp256k1 from "secp256k1";
+import { default as blake } from "blake-hash";
+import * as bs58 from "bs58";
+import toBuffer from "typedarray-to-buffer";
+import { getTxFromInputs } from "../actions/TransactionActions";
+
+export function addressPath(branch, index) {
+  const prefix = "44'/42'/0'/";
+  const i = (index || 0).toString();
+  const b = (branch || 0).toString();
+  return prefix + b + "/" + i;
+}
+
+export function pubkeyToHDPubkey(pubkey, chainCode, isTestnet) {
+  const pk = secp256k1.publicKeyConvert(hexToBytes(pubkey), true); // from uncompressed to compressed
+  const cc = hexToBytes(chainCode);
+  let hdPublicKeyID = hexToBytes("02fda926"); // dpub
+  if (isTestnet) {
+    hdPublicKeyID = hexToBytes("043587d1"); // tpub
+  }
+  const parentFP = hexToBytes("00000000"); // not true but we dont know the fingerprint
+  const childNum = hexToBytes("80000000"); // always first hardened child
+  const depth = 2; // account is depth 2
+  const buff = new Uint8Array(78); // 4 network identifier + 1 depth + 4 parent fingerprint + 4 child number + 32 chain code + 33 compressed public key
+  let i = 0;
+  buff.set(hdPublicKeyID, i);
+  i += 4;
+  buff[i] = depth;
+  i += 1;
+  buff.set(parentFP, i);
+  i += 4;
+  buff.set(childNum, i);
+  i += 4;
+  buff.set(cc, i);
+  i += 32;
+  buff.set(pk, i);
+  const firstPass = blake("blake256").update(Buffer.from(buff)).digest();
+  const secondPass = blake("blake256").update(firstPass).digest();
+  const fullSerialize = Buffer.concat([
+    Buffer.from(buff),
+    secondPass.slice(0, 4)
+  ]);
+  return bs58.encode(fullSerialize);
+}
+
+function writeUint16LE(n) {
+  const buff = new Buffer(2);
+  buff.writeUInt16LE(n, 0);
+  return buff;
+}
+
+function writeUint32LE(n) {
+  const buff = new Buffer(4);
+  buff.writeUInt32LE(n, 0);
+  return buff;
+}
+
+function writeUint64LE(n) {
+  const buff = new Buffer(8);
+  const lower = 0xffffffff & n;
+  // bitshift right (>>>) does not seem to throw away the lower half, so
+  // subtracting.
+  const upper = n - lower;
```

Comment:
wait, this doesnt work. hmm..

Reply:
Division with the max uint32 works correct? What is the correct syntax to bitshift right though? Is it not `n>>32` or `n>>>32`?

Not using `Buffer.writeUint64LE` as this version of Buffer does not seem to have it.